media-video/avidemux/avidemux-9999.ebuild

```diff
@@ -1,113 +1,121 @@
+		CMAKE_USE_DIR="${S}"/${process#*:} BUILD_DIR="${build}" cmake-utils_src_prepare
+	done
 
 	# Fix icon name -> avidemux-2.6.png
-	sed -i -e "/^Icon/ s:${PN}:${PN}-2.6:" ${PN}2.desktop || die "Icon name fix failed."
+	sed -i -e "/^Icon/ s:${PN}:${PN}-2.6:" "${S}"/${PN}2.desktop || die "Icon name fix failed."
 
 	# The desktop file is broken. It uses avidemux2 instead of avidemux3
 	# so it will actually launch avidemux-2.5 if it is installed.
-	sed -i -e "/^Exec/ s:${PN}2:${PN}3:" ${PN}2.desktop || die "Desktop file fix failed."
-	sed -i -re '/^Exec/ s:(avidemux3_)gtk:\1qt4:' ${PN}2.desktop || die "Desktop file fix failed."
+	sed -i -e "/^Exec/ s:${PN}2:${PN}3:" "${S}"/${PN}2.desktop || die "Desktop file fix failed."
+	sed -i -re '/^Exec/ s:(avidemux3_)gtk:\1qt4:' "${S}"/${PN}2.desktop || die "Desktop file fix failed."
 
 	# Fix QA warnings that complain a trailing ; is missing and Application is deprecated.
-	sed -i -e 's/Application;AudioVideo/AudioVideo;/g' ${PN}2.desktop || die "Desktop file fix failed."
+	sed -i -e 's/Application;AudioVideo/AudioVideo;/g' "${S}"/${PN}2.desktop || die "Desktop file fix failed."
 
 	# Now rename the desktop file to not collide with 2.5.
-	mv ${PN}2.desktop ${PN}-2.6.desktop || die "Collision rename failed."
+	mv "${S}"/${PN}2.desktop "${S}"/${PN}-2.6.desktop || die "Collision rename failed."
 
 	# Remove "Build Option" dialog because it doesn't reflect what the GUI can or has been built with. (Bug #463628)
-	sed -i -e '/Build Option/d' avidemux/common/ADM_commonUI/myOwnMenu.h || die "Couldn't remove \"Build Option\" dialog."
+	sed -i -e '/Build Option/d' "${S}"/avidemux/common/ADM_commonUI/myOwnMenu.h || die "Couldn't remove \"Build Option\" dialog."
 }
 
 src_configure() {
-	local mycmakeargs="
-		-DAVIDEMUX_SOURCE_DIR='${S}'
-		$(cmake-utils_use nls GETTEXT)
-		$(cmake-utils_use sdl)
-		$(cmake-utils_use vaapi LIBVA)
-		$(cmake-utils_use vdpau)
-		$(cmake-utils_use video_cards_fglrx XVBA)
-		$(cmake-utils_use xv XVIDEO)
-	"
-
-	if use debug ; then
-		mycmakeargs+=" -DVERBOSE=1 -DCMAKE_BUILD_TYPE=Debug -DADM_DEBUG=1"
-	fi
-
-	processes="buildCli:avidemux/cli"
-	use qt4 && processes+=" buildQt4:avidemux/qt4"
-
-	for process in ${processes} ; do
-		local build="${process%%:*}"
-
-		mkdir "${S}"/${build} || die "Can't create build folder."
-		cd "${S}"/${build} || die "Can't enter build folder."
-		CMAKE_USE_DIR="${S}"/${process#*:} BUILD_DIR="${S}"/${build} cmake-utils_src_configure
-	done
-
 	# Add lax vector typing for PowerPC.
 	if use ppc || use ppc64 ; then
 		append-cflags -flax-vector-conversions
 	fi
 
 	# See bug 432322.
 	use x86 && replace-flags -O0 -O1
+
+	local mycmakeargs=(
+		-DAVIDEMUX_SOURCE_DIR='${S}'
+		-DGETTEXT="$(usex nls)"
+		-DSDL="$(usex sdl)"
+		-DLIBVA="$(usex vaapi)"
+		-DVDPAU="$(usex vdpau)"
+		-DXVBA="$(usex video_cards_fglrx)"
+		-DXVIDEO="$(usex xv)"
+	)
+
+	if use debug ; then
+		mycmakeargs+=( -DVERBOSE=1 -DADM_DEBUG=1 )
+	fi
+
+	for process in ${processes} ; do
```

Comment:
That would work for the loops in src_compile and src_install, since a valid unique `BUILD_DIR` is expanded out in each iteration, but the loop in src_configure requires a unique `CMAKE_USE_DIR` to be expanded in `multibuild_foreach_variant()`.  Sadly, it doesn't support deferred variable expansion ( i.e., `eval "${@}"` instead of `"${@}"`).